src/StateSpace.hpp

```diff
@@ -28,6 +28,38 @@ class StateSpace {
     virtual T intermediateState(const T &source, const T &target, float stepSize) const = 0;
 
     /**
+     * An overloaded version designed for use in adaptive stepsize control.
+     *
+     * @param ascLimit The maximum size that the stepsize can be, even when there are no nearby obstacles
+     */
+    virtual T intermediateState(const T &source, const T &target, float prevStepSize, float ascGrowthRate, float defaultStepSize) const = 0;
```

Comment:
It'd be good to document here what it means to pass a value of zero for `prevStepSize`.